### PR TITLE
Update dotnet samples

### PR DESCRIPTION
**Pull request by pjanotti**
## Why

.NET Samples were not updated for some time.

## What

- Updated auto-instrumentation sample to v0.2.2 replacing smart-gateway with latesst OTel collector.
- Removed the manual instrumentation sample: for the cases that pure manual instrumentation is desired we should point users to OTel. Manual instrumentation integrated with auto-instrumentation already have samples on SFx.NET 

## Tests

- Validated manually for auto-instrumentation, see https://app.signalfx.com/#/apm/traces/69b0ca11cf2a30aa4807c9002114d56a
- Built manual instrumentation sample

### CODE REVIEW

File: dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj [hunk 1]
```diff
@@ -6,7 +6,7 @@
   <ItemGroup>
     <PackageReference Include="Microsoft.Azure.ServiceBus" Version="4.1.3" />
     <PackageReference Include="Microsoft.NET.Sdk.Functions" Version="3.0.3" />
-    <PackageReference Include="SignalFx.Tracing.OpenTracing" Version="0.1.6" />
+    <PackageReference Include="SignalFx.Tracing.OpenTracing" Version="0.1.16" />
```
Comment (RassK) [inline dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj:9]: Should be removed instead?

File: signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml [hunk 2]
```diff
@@ -5,20 +5,22 @@ services:
...
+      - otel-collector
     user: 'root'
   client:
     build: src/ClientExample
     environment:
-      SIGNALFX_SERVICE_NAME: 'example-client'
-      SIGNALFX_ENDPOINT_URL: 'http://smart-gateway:9080/v1/trace'
+      - SIGNALFX_SERVICE_NAME=example-client
+      - SIGNALFX_ENDPOINT_URL=http://otel-collector:9411/api/v2/spans
+      - SIGNALFX_ENV
```
Comment (RassK) [inline signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml:21]: Better to assign some value?

File: dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj (hunk 1 above)
Comment (RassK) [reply dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj:9]: ... although then the example needs to be updated using OT tracer

File: signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml (hunk 2 above)
Comment (pjanotti) [reply signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml:21]: Also a good idea: I listed just so I remember to use an `.env` file with the actual value but adding a value here it makes easy for people to try it directly with backend.

File: dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj (hunk 1 above)
Comment (pjanotti) [reply dotnet-manual-instrumentation/Azure.Functions/Azure.Functions.csproj]: Opting to remove all manual instrumentation samples, see top comment.

File: signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml [hunk 3]
```diff
@@ -27,8 +29,23 @@ services:
     restart: unless-stopped
     ports:
       - "27017:27017"
-  smart-gateway:
-    image: smart-gateway
-    restart: unless-stopped
+  otel-collector:
+    image: otel/opentelemetry-collector-contrib:0.45.0
```
Comment (pellared) [inline signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml:33]: I think we should rather use `quay.io/signalfx/splunk-otel-collector:0.45.0`

It is our recommendation and also should simplify this example (the OOTB config should be good).

Here is a docker-compose example: https://github.com/signalfx/splunk-otel-collector/tree/main/examples/docker-compose

**Review by pellared — APPROVED** [review]

File: signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml (hunk 3 above)
Comment (pjanotti) [reply signalfx-tracing/signalfx-dotnet-tracing/aspnetcore-and-mongodb/docker-compose.yml:33]: I will give it a try later today...

**Pull request merged**